modules/Antivirus/vtsearch.py

```diff
@@ -32,66 +26,84 @@ def check(conf=DEFAULTCONF):
         return False
     if not conf['apikey']:
         return False
+    if not conf['allinfo']:
+        conf['allinfo'] = 0
     return True
 
+
 def scan(filelist, conf=DEFAULTCONF):
-    #Check for key rotation
+    md5_tuples, junk = REQUIRES[0]
     apikey = conf['apikey']
+    all_hashes = [md5[1] for md5 in md5_tuples]
+
+    # Virustotals API limits the number of hashes per request to 25 - sublists of len <=25 are created
+    hash_lists = [all_hashes[i:i + VT_HASH_LIMIT] for i in range(0, len(all_hashes), VT_HASH_LIMIT)]
+
+    # Check for key rotation
     rotkey = False
-    if isinstance(conf['apikey'], list):
-        rotkey = _repeatlist(conf['apikey'])
+    if isinstance(apikey, list):
+        rotkey = _repeatlist(apikey)
         apikey = rotkey.next()
-    results = []
-    md5s, junk = REQUIRES[0]
-    requests = [[]]
-    md5name = {}
-    for fname, md5 in md5s:
-        if len(requests[-1]) == 25:

```

Comment:
Strange that I missed that hash-per-query limit check from the old code given its thorough commenting :) 

I've updated the PR to break all the hashes into <= 25 hash sub-lists.

Reply:
Comments are hard! I'll test it out